maven/JarAssembler.kt

```diff
@@ -34,35 +37,64 @@ class JarAssembler : Callable<Unit> {
     @Option(names = ["--jars"], split = ";")
     lateinit var jars: Array<File>
 
-    val entryNames = mutableSetOf<String>()
+    private val entryNames = mutableSetOf<String>()
+    val entries = HashMap<String, ByteArray>()
+
+    /**
+     * For path "a/b/c.java" inserts "a/" and "a/b/ into `entries`
+     */
+    private fun addDirectories(path: Path) {
```

Comment:
let's call it `preCreateDirectories`?

Reply:
@lolski addressed in https://github.com/vaticle/bazel-distribution/pull/305/commits/a843c9a2834b9693197d6e45ec799274a61284b2